### PR TITLE
Eliminate many cases of JSP print(String)

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
We don't like JSPs rendering raw Strings. Using builders, HTML DOM, HtmlString, tags, and helper methods is easier and ensures consistency.

#### Changes
* Eliminate many cases of JSP print(String). Clear the BVT-A suite from 25,000 to 0 instances and BVT-B from 21,000 to 0 instances.

### CODE REVIEW

File: api/src/org/labkey/api/miniprofiler/MiniProfiler.java [hunk 1]
```diff
@@ -133,29 +134,28 @@ public static void resetSettings(@NotNull User user)
...
+            "  ids:" + ids + ",\n" +
+            "  version:" + version + ",\n" +
+            "  renderPosition:'" + settings.getRenderPosition().getStyle() + "',\n" +
+            "  showTrivial:" + settings.isShowTrivial() + ",\n" +
+            "  trivialMilliseconds:" + settings.getTrivialMillis() + ",\n" +
+            "  showChildrenTime:" + settings.isShowChildrenTime() + ",\n" +
+            "  maxTracesToShow:" + 20 + ",\n" +
+            "  showControls:" + settings.isShowControls() + ",\n" +
+            "  authorized:true,\n" +
+            "  toggleShortcut:'" + (settings.getToggleShortcut() != null ? settings.getToggleShortcut() : "") + "',\n" +
```
Comment (labkey-matthewb) [inline api/src/org/labkey/api/miniprofiler/MiniProfiler.java]: Should probably convert to use jsString()

File: api/src/org/labkey/api/util/PageFlowUtil.java [hunk 2]
```diff
@@ -1548,24 +1540,24 @@ private static void parserSetFeature(XMLReader parser, String feature, boolean b
...
 
-    public static String getAppIncludes(ViewContext context, @Nullable  LinkedHashSet<ClientDependency> resources)
+    public static HtmlString getAppIncludes(ViewContext context, @Nullable  LinkedHashSet<ClientDependency> resources)
     {
         return _getStandardIncludes(context, null, resources, false, false);
     }
 
 
-    public static String getStandardIncludes(ViewContext context, @Nullable LinkedHashSet<ClientDependency> resources, boolean includePostParameters)
+    public static HtmlString getStandardIncludes(ViewContext context, @Nullable LinkedHashSet<ClientDependency> resources, boolean includePostParameters)
```
Comment (labkey-matthewb) [inline api/src/org/labkey/api/util/PageFlowUtil.java:1549]: Nice

File: pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp [hunk 3]
```diff
@@ -155,17 +157,17 @@
...
-                    <option value="0" <%=getSelected("0", successNotifyInterval)%>>every job</option>
-                    <option value="1" <%=getSelected("1", successNotifyInterval)%>>1 hour</option>
-                    <option value="2" <%=getSelected("2", successNotifyInterval)%>>2 hours</option>
-                    <option value="3" <%=getSelected("3", successNotifyInterval)%>>3 hours</option>
-                    <option value="4" <%=getSelected("4", successNotifyInterval)%>>4 hours</option>
-                    <option value="5" <%=getSelected("5", successNotifyInterval)%>>5 hours</option>
-                    <option value="6" <%=getSelected("6", successNotifyInterval)%>>6 hours</option>
-                    <option value="12" <%=getSelected("12", successNotifyInterval)%>>12 hours</option>
-                    <option value="24" <%=getSelected("24", successNotifyInterval)%>>24 hours</option>
+                    <option value="0"<%=selected("0".equals(successNotifyInterval))%>>every job</option>
```
Comment (labkey-matthewb) [inline pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp]: consider:  selected(a,b) for conciseness

File: study/src/org/labkey/study/view/datasets.jsp [hunk 4]
```diff
@@ -58,70 +61,59 @@
...
-            sb.append("<a href=\"").append(datasetURL.replaceParameter("datasetId", String.valueOf(dataset.getDatasetId())));
-            sb.append("\">");
-            sb.append(h(datasetLabel));
-            sb.append("</a></td></tr>\n");
+            tds.add(TR(TD(link(datasetLabel).href(datasetURL.replaceParameter("datasetId", String.valueOf(dataset.getDatasetId()))).clearClasses())));
         }
-        sb.append("    </table>");
 
-        return sb.toString();
+        TABLE(tds.toArray()).appendTo(out);
```
Comment (labkey-matthewb) [inline study/src/org/labkey/study/view/datasets.jsp:116]: would be nice to be able to do <%= TABLE() %>

File: pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp (hunk 3 above)
Comment (labkey-matthewb) [reply pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp]: BUG checked(displaySuccess.equals(""))

Comment (labkey-adam) [reply pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp]: Replaced with `HtmlString.EMPTY_STRING.equals(displaySuccess)`

Comment (labkey-adam) [reply pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp]: Note that the long list of notify interval options will get massively simplified in a follow-on story that replaces all hand-coded select/option elements with SelectBuilder. In this case, we'll just pass in a Map<String, String> and call `selected(successNotifyInterval)` once.

File: study/src/org/labkey/study/view/datasets.jsp (hunk 4 above)
Comment (labkey-adam) [reply study/src/org/labkey/study/view/datasets.jsp:116]: Do you have a suggestion for how we'd achieve this? Does Renderable.toString() produce the rendered output we need?

File: pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp (hunk 3 above)
Comment (labkey-adam) [reply pipeline/src/org/labkey/pipeline/emailNotificationSetup.jsp]: Oh, I guess you already fixed the bug

**Review by labkey-matthewb — APPROVED** [review]

**Pull request merged**